swarm/api/routes/evolution.py

```
@@ -409,10 +411,33 @@ async def apply_evolution_patch_endpoint(
     else:
+        try:
+            patch_id = validate_path_component(request.patch_id, "patch_id")
+        except ValueError as e:
+            raise HTTPException(
+                status_code=400,
+                detail={
+                    "error": "invalid_path_component",
+                    "message": str(e),
+                    "details": {},
+                },
+            )
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)